index.js

```diff
@@ -92,18 +85,15 @@ function render (file, elem, opts, cb) {
 }
 
 function append (file, rootElem, opts, cb) {
-  if (typeof opts === 'function') {
-    cb = opts
-    opts = {}
-  }
+  if (typeof opts === 'function') [cb, opts] = [opts, cb]
```

Comment:
This isn't the same behavior.

if the third parameter is a function, then the fourth will be undefined (options are always the third param in the API, but they're optional).

This would be the same
```js
[cb, opts] = [opts, {}]
```

I think the original code is more readable though. Maybe this would be even better
```js
if (typeof opts === 'function') return append(file, rootElem, undefined, opts)
```

Reply:
I thought since we are doing `if (!opts) opts = {}` just right below it would be fine to just swap the two

that would make it possible to do it both ways
`
render.append(file, rootElem, opts, cb)
render.append(file, rootElem, cb, opts)
`

Reply:
I guess that would eventually lead to the same behavior yeah, but it sacrifices readability a lot IMO

Reply:
@KayleePop You really made my head spin all day today, searched on google like crazy. First asked my brother who is also a senior java/android developer but he could't answer my question. So I turned to StackOverflow for an answer - only so I could explain to you why i swap variables instead of returning a new function. (The words I was looking for where "free variables" and "lexical")

----

IMO, I think both (swap/return) are as equally readable but I tend to go for writing [functions with as few free variables](https://stackoverflow.com/questions/52140468/what-do-you-call-function-that-use-stuff-outside-of-the-function-and-function-is) as possible which should technically be faster since it don't have to look up lexically in the scope. One could also argue that It would also make the stack trace 1 line sorter if it only called a function once and if it where to throw some error

(and last but not least· it would make the final bundle size is smaller. But this so tiny and should not even be a argument up for debatable since it depends on the context of what code you are writing.)

Writing functions that have no free variables at all should technically be well optimized and good for tree shaking (that is just one of my practices i follow - but all developers have different opinions - no way getting over that)

To me this is just one of those "to use semicolon or not to use" or "tab vs space" "2 or 4 spaces" - it just about picking something. It just takes some time to getting used to

If I failed to convince you it's better to use less "free variables" over readable code I will gladly change this to `return append(...args)` if it makes it you happy, for me it dose not really matter.

Reply:
just for the heck of it: An example of what it means to use closed-over variable from the outer scope and what the minifier would be able to accomplish 

```js
module.export = function get(a) {
  if (a < 0) return get(a * -1)
  return a
}
// minified
module.export=function b(a){return 0>a?b(-1*a):a}
```

```js
module.export = function get(a) {
  if (a < 0) a = a * -1
  return a
}
// minified
module.export=a=>a<0?-1*a:a
```
^ you get a anonymous function :P

Reply:
IDK, this sounds like premature optimization to me.

I think the original way of doing it (before this PR) is just as readable as returning the new call of the function. I was mostly just giving another option (the way I suggested is used in webtorrent).

The swapping is confusing though, because in that scenario, the fourth parameter isn't defined or important. When you swap the two it makes it seem like the value of the parameter `cb` is important when it's not, but IDK maybe that's just me.

I still think this is premature optimization, but I'm not sure if this example would be considered a free variable because all the variables are still within the scope of the same function, they're just another layer deep. If you only imported this function it would still work because it just calls itself again.